modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java

```diff
@@ -97,7 +97,7 @@ public GoClientCodegen() {
         typeMapping.put("boolean", "bool");
         typeMapping.put("string", "string");
         typeMapping.put("UUID", "string");
-        typeMapping.put("date", "time.Time");
+        typeMapping.put("date", "SwaggerDateType");
```

Comment:
String worked with minimal changes for the user. It can be converted by using
`dateObject, err := time.Parse("2006-01-02", dateString)`